pyctcdecode/language_model.py

```diff
@@ -379,6 +387,66 @@ def load_from_dir(cls, filepath: str) -> "LanguageModel":
         kenlm_model = kenlm.Model(filenames["kenlm"])
         return cls(kenlm_model, unigrams, **json_attrs)
 
+    @classmethod
+    def load_from_hf_hub(cls, pretrained_path: str, kenlm_filename: str = "kenLM.arpa", cache_dir: Optional[str] = None, **kwargs):
+        """Class method to load model from https://huggingface.co/
+
+        Args:
+            pretrained_path: string, the `model id` of a pretrained model hosted inside a model
+                repo on https://huggingface.co/. Valid model ids can be namespaced under a user or
+                organization name, like ``kensho/5gram-spanish-kenLM``. For more information, please
+                take a look at https://huggingface.co/docs/hub/main .
+            kenlm_filename: file name of the kenlm model as expected to be saved on https://huggingface.co/. Defaults to "kenlm.arpa".
+            cache_dir: path to where the language model should be downloaded and cached.
+        """
+        from . import __version__ as VERSION, __package_name__ as LIBRARY_NAME
+
+        CACHE_DIRECTORY = cache_dir or os.path.join(Path.home(), ".cache", LIBRARY_NAME)
+
+        try:
+            from huggingface_hub import hf_hub_download
+        except ImportError:
+            raise ImportError(
+                "You need to install huggingface_hub to use `load_from_hf_hub`. "
+                "See https://pypi.org/project/huggingface-hub/ for installation."
+            )
+
+        # download and cache kenLM model
```

Comment:
download each file seperatly

Reply:
is there no way to download the whole thing as a tarball? 
if not, there's the potential to download a kenlm file and a unigrams file that is incompatible with it

Reply:
I think we could find a solution to download a tar ball instead if you prefer!

IMO there would be the following downsides though:
1) Users cannot verify the files online on the hub. E.g. right now one can easily check the saved configs of the Language Model here:
https://huggingface.co/kensho/dummy_full_language_model/blob/main/attrs.json
=> This would not be possible if a `.tar` file is saved instead
2) the downloaded tar file would be cached instead of each of the three files. This means that after caching, the tar file would be untared every time `load_from_hf_hub` is called. I don't think it's possible to download a tar ball and untar it and then cache the whole directory.

Especially 2.) does not give a great user experience IMO.

Reply:
Also cc'ing @osanseviero here - do we have examples where whole folders are saved as a `.tar` ball on the hub?